### PR TITLE
feature(mining): Enable mining RPCs by default in production builds

### LINKED ISSUE #7361: Make getblocktemplate-rpcs a default feature
### Post-Merge TODOs

- [x] Describe this newly stabilised feature in the CHANGELOG
- [x] Arborist announcement
- [ ] Forum update

### Motivation

Once we're happy to support the getblocktemplate RPCs, we can turn them on in all Zebra builds, by adding that feature to the default feature list:
https://github.com/ZcashFoundation/zebra/blob/31426f3b0e38bb681df3b82a38dad07405a876a5/zebrad/Cargo.toml#L53-L55

### Specifications

https://doc.rust-lang.org/cargo/reference/features.html#the-default-feature

### Complex Code or Requirements

This is a well-supported Cargo configuration.

### Testing

Some tests might need updates if they assume `getblocktemplate-rpcs` will always be disabled during the test. We should probably delete or disable those tests (ticket #7362).

### PR DESCRIPTION
**Pull request by teor2345**
## Post-Merge TODOs

- [ ] Describe this newly stabilised feature in the CHANGELOG
- [x] Arborist announcement
- [ ] Forum update

## Motivation

We want to enable mining by default in production builds.

Close #7361
Depends-On: #7744

## Solution

- Enable the feature by default
- Update the documentation to remove custom builds and docker images to enable the feature
- Update the README to point to the Docker mining instructions
  - Consistently use ZF funding stream addresses in examples

### Testing

- Always set the miner address in tests based on the network

This is a very verbose change, but all it does is pass the network from all the tests down to the base test config method, then set the miner address for mainnet or testnet there.

## Review

This is a pretty straightforward change. It doesn't matter which release it goes in.

I'd like extra documentation cleanups to go in another PR or ticket, that's what we're doing with test cleanups.

### Reviewer Checklist

  - [ ] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [ ] Are the PR labels correct?
  - [ ] Does the code do what the ticket and PR says?
    - [ ] Does it change concurrent code, unsafe code, or consensus rules?
  - [ ] How do you know it works? Does it have tests?

## Follow Up Work

These follow-up changes will be made in separate PRs:
- #7362
- any other documentation changes
- any other cleanups

### CODE REVIEW

**Comment by teor2345** [comment]
This testnet failure is a blocker for merging this change:

```
   Message:  assertion `left == right` failed: incorrect miner address config: t3W6Tc1o7J93CWotRGpxgmyMKiuEMA3JpUh network.network Testnet and miner address network Mainnet must match
     left: Mainnet
    right: Testnet
   Location: /opt/zebrad/zebra-rpc/src/server.rs:155
```
https://github.com/ZcashFoundation/zebra/actions/runs/6503509916/job/17664399693?pr=7740#step:9:309

It looks like the test mining config code was never updated for testnet jobs, I'll fix that on Monday.

**Comment by teor2345** [comment]
Failed due to #7744, I restarted CI.

**Comment by teor2345** [comment]
It seems like the fix in PR #7744 might be needed for CI to pass reliably on this PR.

File: README.md [hunk 1]
```diff
@@ -128,11 +128,15 @@ In future releases, the `progress_bar = "summary"` config will show a few key me
...
+##### Configuring Mining
+
+Zebra can be configured for mining by passing a `MINER_ADDRESS` and port mapping to Docker.
+See the [mining support docs](https://zebra.zfnd.org/user/mining-docker.html) for more details.
+
 ##### Custom Build Features
 
 You can also build Zebra with additional [Cargo features](https://doc.rust-lang.org/cargo/reference/features.html#command-line-feature-options):
 
-- `getblocktemplate-rpcs` for [mining support](https://zebra.zfnd.org/user/mining.html)
```
Comment (arya2) [inline README.md:135]: Do we want to wait until this change has been released before updating the readme?

File: book/src/user/mining.md [hunk 2]
```diff
@@ -1,29 +1,24 @@
...
 - [Running zebra](#running-zebra)
 - [Testing the setup](#testing-the-setup)
 - [Run a mining pool](#run-a-mining-pool)
 
-## Download and build Zebra for mining
+## Download Zebra
 [#download-and-build-zebra]: #download-and-build-zebra
 
-Before installing please make sure you have the [Zebra dependencies](https://github.com/ZcashFoundation/zebra#build-instructions) in your OS.
+The easiest way to run Zebra for mining is [our docker images](https://zebra.zfnd.org/user/mining-docker.html).
```
Comment (arya2) [inline book/src/user/mining.md]: Nitpick:

```suggestion
The easiest way to run Zebra for mining is with [our docker images](https://zebra.zfnd.org/user/mining-docker.html).
```

File: zebra-rpc/src/config/mining.rs [hunk 3]
```diff
@@ -46,9 +46,6 @@ impl Config {
     /// This is used to ignore the mining section of the configuration if the feature is not
     /// enabled, allowing us to log a warning when the config found is different from the default.
     pub fn skip_getblocktemplate(&self) -> bool {
-        #[cfg(feature = "getblocktemplate-rpcs")]
-        return false;
-        #[cfg(not(feature = "getblocktemplate-rpcs"))]
-        return true;
+        !cfg!(feature = "getblocktemplate-rpcs")
```
Comment (arya2) [inline zebra-rpc/src/config/mining.rs:49]: Optional/Nitpick/Subjective:

```suggestion
        cfg!(not(feature = "getblocktemplate-rpcs"))
```

**Review by arya2 — DISMISSED** [review]
Looks good, may need a rebase.

File: README.md (hunk 1 above)
Comment (teor2345) [reply README.md:135]: No, we don't usually do that.

File: zebra-rpc/src/config/mining.rs (hunk 3 above)
Comment (teor2345) [reply zebra-rpc/src/config/mining.rs:49]: I think it's clear either way

**Review by arya2 — APPROVED** [review]

**Pull request merged**